snapshot/snapshots/fake_filer.go

```diff
@@ -88,3 +92,67 @@ func (c *initingCheckouter) Checkout(id string) (snapshot.Checkout, error) {
 		id:   id,
 	}, nil
 }
+
+// Creates a new Ingester that will copy ingested paths to its own tmp snapshots dir.
+func MakeTempIngester(tmp *temp.TempDir) snapshot.Ingester {
+	return &tempIngester{tmp: tmp, snapshots: make(map[string]string)}
+}
+
+type tempIngester struct {
+	tmp       *temp.TempDir
+	snapshots map[string]string
+}
+
+func (t *tempIngester) Ingest(path string) (id string, err error) {
+	_, err = os.Stat(path)
+	if err != nil {
+		return "", nil
+	}
+
+	var snapdir *temp.TempDir
+	snapdir, err = t.tmp.TempDir("snap-")
+	if err != nil {
+		return "", err
+	}
+	err = exec.Command("cp", "-rf", path, snapdir.Dir).Run()
+	if err != nil {
+		return "", err
+	}
+
+	id = strconv.Itoa(len(t.snapshots))
+	t.snapshots[id] = snapdir.Dir
+	return
+}
+
+// Creates a filer that copies ingested paths in and then back out for checkouts.
+func MakeTempFiler(tmp *temp.TempDir) snapshot.Filer {
+	return &tempFiler{tempIngester: MakeTempIngester(tmp).(*tempIngester)}
+}
+
+type tempFiler struct {
```

Comment:
I know why it's worth splitting out the interfaces Ingester and Checkouter, but it might be fine to just have the one impl tempFiler (instead of both tempFiler and tempIngester)